core/imds/src/main/java/software/amazon/awssdk/imds/MetadataResponse.java

```diff
@@ -0,0 +1,187 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.imds;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.protocols.jsoncore.JsonNode;
+import software.amazon.awssdk.protocols.jsoncore.JsonNodeParser;
+
+/**
+ * The class is used for Response Handling and Parsing the metadata fetched by the get call in the {@link Ec2Metadata} interface.
+ * Metadata is stored in the instance variable <b>body</b>. The class provides convenience methods to the users to parse the
+ * metadata as a String, List and to parse the metadata in the JsonResponse according to the key.
+ */
+@SdkPublicApi
+public class MetadataResponse {
+
+    private static final Logger log = LoggerFactory.getLogger(MetadataResponse.class);
+
+    private static final JsonNodeParser JSON_NODE_PARSER = JsonNode.parser();
+    
+    private final String body;
+
+    public MetadataResponse(String body) {
+        this.body = body;
+    }
+
+    /**
+     * Returns the Metadata Response body as a String. This method can be used for parsing the retrieved
+     * singular metadata from IMDS.
+     *
+     * @return String Representation of the Metadata Response Body.
+     *
+     * <p>
+     * Example:
+     * <pre>
+     * {@code
+     *
+     * Ec2Metadata ec2Metadata = Ec2Metadata.builder().build();
+     * MetadataResponse metadataResponse = client.get("/latest/meta-data/ami-id");
+     * String response = metadataResponse.asString();
+     *  }
+     *  </pre>
+     */
+    public String asString() {
+        return body;
+
+    }
+
+    /**
+     * Parses the response String into a list of Strings split by delimiter ("\n"). This method can be used for parsing the
+     * list-type metadata from IMDS.
+     *
+     * @return List Representation of the Metadata Response Body.
+     *
+     * <p>
+     * Example:
+     * <pre>
+     * {@code
+     *
+     * Ec2Metadata ec2Metadata = Ec2Metadata.builder().build();
+     * MetadataResponse metadataResponse = client.get("/latest/meta-data/ancestor-ami-ids");
+     * List<String>response = metadataResponse.asList();
+     * }
+     * </pre>
+     */
+    public List<String> asList() {
+
+        if (null != body && body.contains("\n"))  {
+
+            return Arrays.asList(body.split("\n"));
+        }
+
+        return Collections.emptyList();
```

Comment:
If it's a single line, should we return that single line in a collection instead of not returning anything?